lib/hyacinth/batch_import/index_prerequisite_map.rb

```diff
@@ -0,0 +1,237 @@
+# frozen_string_literal: true
+
+module Hyacinth
+  module BatchImport
+    module IndexPrerequisiteMap
+      # This method reads through the data in the given csv_file and generates a prerequisite map that
+      # determines the order of processing, and which processing operations should be blocked by other
+      # prerequisite operations.  The returned array looks something like this:
+      # {
+      #   4 => [3],    # 4 requires 3
+      #   3 => [2],    # 3 requires 2
+      #   2 => [1, 5], # 2 requires 1 and 5
+      #   8 => [7],    # 8 requires 7
+      #   7 => [2],    # 7 requires 2
+      #   10 => [9]    # 10 requires 9
+      # }
+      def self.generate(csv_file)
+        # First, we'll generate two maps:
+        identifiers_to_row_numbers = {}
+        row_numbers_to_parent_identifiers = {}
+
+        ::BatchImport.csv_file_to_hierarchical_json_hash(csv_file) do |json_hash_for_row, csv_row_number|
+          identifiers = json_hash_for_row['identifiers'] || []
```

Comment:
Yeah, that's exactly what it is -- the generic "identifiers" array attribute on a DigitalObject.

Reply:
This might not be something we need to worry about, but if someone adds a uid that points at a different object in Hyacinth, in theory bad things would happen? Like information would get mapped to the wrong object?

Reply:
Yeah, you're right, I don't think we need to worry about that.  I don't think we can handle that kind of user mistake since if they put the wrong _uid field value for a row then they're basically saying "this is the UID for the record I want to update for this row" instead of their intended one.